### PR TITLE
Feature/triggered by

### PR DESCRIPTION
**Pull request by SevenOnes**

### CODE REVIEW

File: thundra/opentracing/tracer.py [hunk 1]
```diff
@@ -24,6 +24,7 @@ def __init__(self, recorder=None, scope_manager=None):
         self.global_span_order = 0
         self.test_xray_traces = []
```
Comment (serkan-ozal) [inline thundra/opentracing/tracer.py]: there should not be `test_xray_traces` here? I hope that this is redundant or left over

File: thundra/opentracing/tracer.py [hunk 2]
```diff
@@ -24,6 +24,7 @@ def __init__(self, recorder=None, scope_manager=None):
...
         ThundraTracer.__instance = self
+        self.function_name = None
```
Comment (serkan-ozal) [inline thundra/opentracing/tracer.py]: It is better we should not have something specific to function here. I don't know the requirement of having `function_name` here but it should not be handled through tracer

File: thundra/integrations/mysql.py [hunk 3]
```diff
@@ -32,7 +33,10 @@ def inject_span_info(self, scope, cursor, connection, _args, _kwargs, response,
             constants.SpanTags['DB_STATEMENT']: query,
             constants.SpanTags['DB_STATEMENT_TYPE']: operation.upper(),
             constants.SpanTags['TRIGGER_DOMAIN_NAME']: "AWS-Lambda",
-            constants.SpanTags['TRIGGER_CLASS_NAME']: "API"
+            constants.SpanTags['TRIGGER_CLASS_NAME']: "API",
```
Comment (serkan-ozal) [inline thundra/integrations/mysql.py]: setting `TOPOLOGY_VERTEX'` tag to `true`?

File: thundra/integrations/postgre.py [hunk 4]
```diff
@@ -42,7 +43,10 @@ def inject_span_info(self, scope, cursor, connection, _args, _kwargs, response,
             constants.SpanTags['DB_STATEMENT']: query,
             constants.SpanTags['DB_STATEMENT_TYPE']: operation.upper(),
             constants.SpanTags['TRIGGER_DOMAIN_NAME']: "AWS-Lambda",
-            constants.SpanTags['TRIGGER_CLASS_NAME']: "API"
+            constants.SpanTags['TRIGGER_CLASS_NAME']: "API",
```
Comment (serkan-ozal) [inline thundra/integrations/postgre.py]: setting `TOPOLOGY_VERTEX'` tag to `true`?

File: thundra/integrations/redis.py [hunk 5]
```diff
@@ -42,6 +42,9 @@ def inject_span_info(self, scope, wrapped, instance, args, kwargs, response, exc
             Constants.RedisTags['REDIS_COMMAND_TYPE']: command_type,
             Constants.RedisTags['REDIS_COMMAND_ARGS']: [str(arg) for arg in args],
             Constants.RedisTags['REDIS_COMMAND']: self.getCommandType(command_type),
+            Constants.SpanTags['TRIGGER_OPERATION_NAMES']: [scope.span.tracer.function_name],
```
Comment (serkan-ozal) [inline thundra/integrations/redis.py]: setting `TOPOLOGY_VERTEX'` tag to `true`?

File: thundra/integrations/requests.py [hunk 6]
```diff
@@ -38,6 +38,9 @@ def inject_span_info(self, scope, wrapped, instance, args, kwargs, response, exc
             constants.HttpTags['HTTP_PATH']: path,
             constants.HttpTags['HTTP_HOST']: host,
             constants.HttpTags['QUERY_PARAMS']: query,
+            constants.SpanTags['TRIGGER_OPERATION_NAMES']: [scope.span.tracer.function_name],
```
Comment (serkan-ozal) [inline thundra/integrations/requests.py]: setting `TOPOLOGY_VERTEX'` tag to `true`?

File: thundra/integrations/botocore.py [hunk 7]
```diff
@@ -50,7 +50,7 @@ def inject_span_info(self, scope, wrapped, instance, args, kwargs, response,
 
         scope.span.domain_name = Constants.DomainNames['DB']
         scope.span.class_name = Constants.ClassNames['DYNAMODB']
-        scope.span.operation_name = 'dynamodb: ' + str(self.request_data['TableName'])
+        scope.span.operation_name = str(self.request_data['TableName'])
```
Comment (serkan-ozal) [inline thundra/integrations/botocore.py]: what if there is no `TableName` in the request? For example, when the request is for list tables

File: thundra/integrations/botocore.py [hunk 8]
```diff
@@ -205,15 +223,14 @@ def getRequestType(self, string):
...
+                         exception):
 
         operation_name, request_data = args
         self.streamName = request_data['StreamName']
 
         scope.span.domain_name = Constants.DomainNames['STREAM']
         scope.span.class_name = Constants.ClassNames['KINESIS']
-        scope.span.operation_name = self.getRequestType(operation_name)
-
+        scope.span.operation_name = self.streamName
```
Comment (serkan-ozal) [inline thundra/integrations/botocore.py]: what if there is no `streamName` in the request? For example, when the request is for list streams

File: thundra/integrations/botocore.py [hunk 9]
```diff
@@ -243,14 +266,14 @@ def getRequestType(self, string):
...
-                 exception):
+                         exception):
 
         operation_name, request_data = args
         self.deliveryStreamName = request_data['DeliveryStreamName']
 
         scope.span.domain_name = Constants.DomainNames['STREAM']
         scope.span.class_name = Constants.ClassNames['FIREHOSE']
-        scope.span.operation_name = self.getRequestType(operation_name)
+        scope.span.operation_name = self.deliveryStreamName
```
Comment (serkan-ozal) [inline thundra/integrations/botocore.py]: what if there is no `deliveryStreamName` in the request? For example, when the request is for list streams

File: thundra/integrations/botocore.py [hunk 10]
```diff
@@ -280,14 +309,14 @@ def getRequestType(self, string):
...
-                 exception):
+                         exception):
 
         operation_name, request_data = args
         self.bucket = request_data['Bucket']
 
         scope.span.domain_name = Constants.DomainNames['STORAGE']
         scope.span.class_name = Constants.ClassNames['S3']
-        scope.span.operation_name = self.getRequestType(operation_name)
+        scope.span.operation_name = self.bucket
```
Comment (serkan-ozal) [inline thundra/integrations/botocore.py]: what if there is no `Bucket` in the request? For example, when the request is for list buckets

File: thundra/integrations/botocore.py [hunk 11]
```diff
@@ -321,13 +356,13 @@ def getRequestType(self, string):
...
     def inject_span_info(self, scope, wrapped, instance, args, kwargs, response,
-                 exception):
+                         exception):
 
         operation_name, request_data = args
         self.lambdaFunction = request_data.get('FunctionName', '')
         scope.span.domain_name = Constants.DomainNames['API']
         scope.span.class_name = Constants.ClassNames['LAMBDA']
-        scope.span.operation_name = self.getRequestType(operation_name)
+        scope.span.operation_name = self.lambdaFunction
```
Comment (serkan-ozal) [inline thundra/integrations/botocore.py]: what if there is no FunctionName in the request? For example, when the request is for list functions

File: thundra/opentracing/tracer.py (hunk 2 above)
Comment (SevenOnes) [reply thundra/opentracing/tracer.py]: I mimicked this from NodeJS agent. It is used to carry function name to integrations.

**Pull request closed without merging**